core/src/main/java/com/nuodb/migrator/globalStore/GlobalStore.java

```diff
@@ -0,0 +1,97 @@
+
+public class GlobalStore {
+
+    private static GlobalStore instance;
+    private String state;
+
+    private Table table;
+
+    HashMap<String, Boolean> generatedValues = new HashMap<>();
+
+    public void put(String key, boolean value) {
+        generatedValues.put(key, value);
+    }
+
+    public boolean get(String key) {
+        return generatedValues.get(key);
+    }
+
+    private Collection<String> columns = Lists.newArrayList();
+
+    private GlobalStore() {
+        // Private constructor to enforce singleton pattern
+    }
+
+    public static synchronized GlobalStore getInstance() {
+        if (instance == null) {
+            instance = new GlobalStore();
+        }
+        return instance;
+    }
+
+    public synchronized String getState() {
+        return state;
+    }
+
+    public void setState(String state) {
+        this.state = state;
+
+    }
+
+    // helper function to create an alter statement to execute after loading the
+    // data, takes the value from hashmap and compares it.
+    public String alterScript(LoadTable table) {
+        try {
+            StringBuilder str = new StringBuilder();
+            for (Column col : table.getTable().getColumns()) {
+                if (col.getSequence() != null) {
+                    str.append("ALTER TABLE ");
+                    str.append(table.getTable().getSchema().getName());
+                    str.append(".");
+                    str.append(table.getTable().getName());
+                    str.append(" MODIFY ");
+                    str.append(col.getName());
+                    str.append(' ');
+                    str.append(col.getJdbcType().getTypeName());
+                    str.append(' ');
+                    String key = col.getSequence().getName().toString();
+                    for (Map.Entry<String, Boolean> entry : generatedValues.entrySet()) {
+                        String entryInHashMap = entry.getKey();
+                        entryInHashMap.replaceAll("^\"|\"$", "");
+                        entryInHashMap = entryInHashMap.replaceAll("^\"|\"$", "");
+                        boolean valueInHashMap = entry.getValue();
+                        if (generatedValues != null) {
+                            if (key.equals(entryInHashMap)) {
+                                if (valueInHashMap) {
+                                    str.append("GENERATED ALWAYS AS IDENTITY");
+                                    return str.toString();
+                                } else {
+                                    return null;
+                                }
+                            } else {
+                                continue;
+                            }
+                        } else
+                            return null;
+                    }
+                }
+            }
+            return null;
+        } catch (Exception e) {
+            System.out.println("Caught Exception in GlobalStore");
```

Comment:
Is correct to just suppress the exception? Do we want to print the stacktrace? Should we log the error rather than printing to stdout?

We should at least handle `InterruptedException` correctly.

We should at least handle `InterruptedException` correctly, by resetting the interrupted status so that any thread that executes a blocking call will throw `InterruptedException` again. That is how `Thread.interrupt()` is supposed to be processes so that the thread eventually exits (there is no forcible thread interrupt in Java).

If `InterruptedException` is not thrown, then you should make the `catch` clause more specific by explicitly declaring all of the checked exceptions that might be thrown.